dandi/consts.py

```diff
@@ -100,6 +102,7 @@ class DandiInstance:
     name: str
     gui: str | None
     api: str
+    is_private: Optional[bool] = None
```

Comment:
unclear to me what `None` would mean different from `False` or what this generally is about.  If `None` is the same semantic as `False` -- let's just make `bool`. Not sure if any annotation to be used with those attributes, at least non-officially, could be done via docstring (https://github.com/python/cpython/issues/82582) e.g.

```suggestion
    is_private: Optional[bool] = None
    """Whether instance accessible to public (`True`), or requires authentication (`False`)""" 
```